Library/Homebrew/formula_installer.rb

```diff
@@ -1259,13 +1299,14 @@ def fetch
     self.class.fetched << formula
   end
 
+  sig { returns(T.any(Bottle, Resource, Resource::Local)) }
   def downloadable
     if (bottle_path = formula.local_bottle_path)
       Resource::Local.new(bottle_path)
     elsif pour_bottle?
-      formula.bottle
+      T.must(formula.bottle)
     else
-      formula.resource
+      T.must(formula.resource)
```

Comment:
@reitermarkus does this look valid to you? The two call sites of this method assume a non-nil return value.

Reply:
`bottle` looks correct given the condition. I assume `resource` is only ever `nil` if there is no `url`, so should be good too.